### PR TITLE
Issue #6331: Add isDefault bool to TopSiteEntity and APIs for flagging default top sites

### LINKED ISSUE #6331: Add isDefault bool to TopSiteEntity and APIs for flagging default top sites
Fenix wants to be able to have "default" top sites distinguishable at the db level. 

Also helpful would be changing the consumer API to have `getDefaultTopSites` and `getNonDefaultTopSites`.

See https://github.com/mozilla-mobile/fenix/issues/9286

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #6331. 

@Amejia481 Just to give you some context about the changes in this PR:

In Fenix, we have added 3 default top sites (Pocket, Wikipedia, YouTube) along with telemetry. 
It is not enough to just have a hardcoded list of these 3 default top sites since we can't differentiate from the user added top sites and top sites migrated from fennec.

We want to add this isDefault boolean to each TopSiteEntity so that in telemetry we can easily just query if the top site is a default top site added by the application. On the front end, we can also sort all the isDefault top sites to the beginning when we are showing the list of top sites.

In this PR, we did the following:

- Add isDefault bool to TopSiteEntity
- Migrate the DB to account for this new property
- Migrate the 3 default top sites to have its isDefault bool set to true

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by gabrielluong** [comment]
@ekager When you have a chance, please take a scan of the PR and let me know if you have any feedback. In the migration of the database schema, we retroactively go back and set `isDefault` for the default top sites that you added. In Fenix, you would only need to add `isDefault = true` when calling `addTopSite`. In terms of ordering, I am wondering as an alternative to the consumer APIs you recommended -- we could potentially do `getTopSites` and sort/filter based on `isDefault`.

My hope as a next step is to add a `position` Integer parameter in the future so we can more actively control the ordering of the top sites and allow for further customization of the order of the top sites.

**Comment by ekager** [comment]
Migration looks great 🙌🏻 thank you for that! No other feedback, this all makes sense to me.

**Comment by gabrielluong** [comment]
> Migration looks great 🙌🏻 thank you for that! This all makes sense to me, but would it also be helpful to add fun to `TopSiteDao` for getting the default top sites via a DB query instead of me having to scan every top site for `isDefault` after getting them all?

@ekager Just wanted to understand your use case a bit more about having a separate method for `getDefaultTopSites`. I believe the use case you are thinking about is for constructing the proper order of items in the AdapterView where we have the default top sites at the beginning. An alternative to adding a new method would be to do `getTopSites().sortBy { !it.isDefault }`.

My only worry was that we added this method and it turns out we didn't need it. 

In future iterations where we look at adding reordering of top sites, we would add `position` Integer into the `TopSiteEntity` and query for `getTopSites()` should always return sorted by `position`.

**Comment by gabrielluong** [comment]
@Amejia481 It will be helpful to read the first comment for more context about this PR and what we're trying to achieve.

**Comment by ekager** [comment]
My thought for a use case would be sending the default top site telemetry if we needed to send the size, but just reread [the telemetry ticket](https://github.com/mozilla-mobile/fenix/issues/8125) and it sounds like we don't actually need to send how many defaults a user has still. Nevermind!

**Comment by gabrielluong** [comment]
@Amejia481 This should be ready for review!

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSite.kt [hunk 1]
```diff
@@ -22,4 +22,9 @@ interface TopSite {
      * The URL of the top site.
      */
     val url: String
+
+    /**
+     * Whether or not the top is a default top site (added as a default by the application).
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSite.kt:27]: Fixed.

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/db/TopSiteEntity.kt [hunk 2]
```diff
@@ -23,6 +23,9 @@ internal data class TopSiteEntity(
     @ColumnInfo(name = "url")
     var url: String,
 
+    @ColumnInfo(name = "isDefault")
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/db/TopSiteEntity.kt:26]: Fixed.

**Comment by Amejia481** [comment]
bors r=ekager,Amejia481

**Pull request merged**